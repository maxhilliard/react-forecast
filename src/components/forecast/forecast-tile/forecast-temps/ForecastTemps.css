.forecastTemps {
    composes: forecastTemps from '../ForecastTile.css';
    font-size: 12px;
    margin: 0;
}

.textElem {
    margin: 0;
    text-align: left;
}