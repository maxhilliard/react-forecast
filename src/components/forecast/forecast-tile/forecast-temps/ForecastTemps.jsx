import React from 'react';
import PropTypes from 'prop-types';

import styles from './ForecastTemps.css';

const tempUnit = 'Â°C';

const ForecastTemps = ({ temp }) => {
    const { min: minTemp, max: maxTemp } = temp;

    return (
        <div className={styles.forecastTemps}>
            <p className={styles.textElem}>
                {`Min: ${minTemp}${tempUnit}`}
            </p>
            <p className={styles.textElem}>
                {`Max: ${maxTemp}${tempUnit}`}
            </p>
        </div>
    );
};

ForecastTemps.propTypes = {
    temp: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
    }).isRequired,
};

export default ForecastTemps;
