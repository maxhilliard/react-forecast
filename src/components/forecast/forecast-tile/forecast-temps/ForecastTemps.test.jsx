import React from 'react';
import { shallow } from 'enzyme';

import ForecastTemps from './ForecastTemps';

describe('ForecastTemps', () => {
    describe('snapshot', () => {
        it('should match snapshot', () => {
            const mockProps = {
                temp: {
                    min: 12,
                    max: 21,
                },
            };

            const wrapper = shallow(<ForecastTemps {...mockProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
