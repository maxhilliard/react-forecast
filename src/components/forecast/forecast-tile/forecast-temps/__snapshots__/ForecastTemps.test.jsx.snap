// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastTemps snapshot should match snapshot 1`] = `
<div
  className="forecastTemps"
>
  <p
    className="textElem"
  >
    Min: 12°C
  </p>
  <p
    className="textElem"
  >
    Max: 21°C
  </p>
</div>
`;
